ako-gateway-api/lib/lib.go

```diff
@@ -79,3 +80,42 @@ func FindListenerStatusByName(name string, status []gatewayv1.ListenerStatus) in
 	}
 	return -1
 }
+
+func FindPortName(serviceName, ns string, servicePort int32, key string) string {
+	// Query the service and obtain the port name
+	svcObj, err := utils.GetInformers().ServiceInformer.Lister().Services(ns).Get(serviceName)
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: error while fetching service object: %s", key, err)
+		return ""
+	}
+	for _, port := range svcObj.Spec.Ports {
+		// Iterate the ports and find the match for targetPort
+		if servicePort == port.Port {
+			utils.AviLog.Debugf("key: %s, msg: Found port name %s for Port: %v", key, port.Name, servicePort)
+			return port.Name
+		}
+	}
+	utils.AviLog.Warnf("key: %s, msg: Port name not found in service obj: %v", key, svcObj)
+	return ""
+}
+
+func FindTargetPort(serviceName, ns string, svcPort int32, key string) intstr.IntOrString {
+	// Query the service and obtain the targetPort
+	svcObj, err := utils.GetInformers().ServiceInformer.Lister().Services(ns).Get(serviceName)
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: error while fetching service object: %s", key, err)
+		return intstr.IntOrString{}
+	}
+	if svcObj.Spec.Type == "NodePort" {
+		// Service of type NodePorts are not supported with tagertPort info. In such a case, the ports in the ingress must be strings
+		return intstr.IntOrString{}
+	}
+	for _, port := range svcObj.Spec.Ports {
+		// Iterate the ports and find the match for targetPort
+		if svcPort == port.Port {
```

Comment:
@aaha97 While defining service, user can specify `portName` or `portNumber` so we should check both of it while returning targetPort for that service.